dialogue-clients/src/main/java/com/palantir/dialogue/clients/ReloadingClientFactory.java

```diff
@@ -126,97 +127,76 @@ public Channel getChannel(String serviceName) {
+                                    .uris(ImmutableList.of(uri))
+                                    .build())
+                            .map(singleUriServiceConf ->
+                                    cache.getNonReloadingChannel(params, singleUriServiceConf, channelName))
+                            .collect(ImmutableList.toImmutableList());
                 });
-                return new AbstractStickyChannels() {
-                    @Override
-                    public Channel getSingleHostChannel() {
-                        return alwaysThrowing;
-                    }
-                };
-            }
 
-            ServiceConfiguration serviceConfiguration =
-                    ServiceConfigurationFactory.of(block).get(serviceName);
+        Refreshable<Supplier<Channel>> bestSupplier = perHostDialogueChannels.map(singleHostChannels -> {
+            if (singleHostChannels.isEmpty()) {
+                AlwaysThrowingChannel alwaysThrowing = new AlwaysThrowingChannel(() -> new SafeIllegalStateException(
+                        "Service not configured (config block not present)", SafeArg.of("serviceName", serviceName)));
+                return () -> alwaysThrowing;
+            }
 
-            if (block.services().get(serviceName).uris().size() == 1) {
-                DialogueChannel singleUriChannel =
-                        cache.getNonReloadingChannel(params, serviceConfiguration, channelName);
-                return new AbstractStickyChannels() {
-                    @Override
-                    public Channel getSingleHostChannel() {
-                        return singleUriChannel;
-                    }
-                };
-            } else {
-                ImmutableList<DialogueChannel> singleUriChannels = serviceConfiguration.uris().stream()
-                        .map(uri -> ServiceConfiguration.builder()
-                                .from(serviceConfiguration)
-                                .uris(ImmutableList.of(uri))
-                                .build())
-                        .map(singleUriServiceConf ->
-                                cache.getNonReloadingChannel(params, singleUriServiceConf, channelName))
-                        .collect(ImmutableList.toImmutableList());
-
-                StickyEndpointChannels stickyEndpointChannels = StickyEndpointChannels.builder()
-                        .channels(singleUriChannels)
-                        .channelName(channelName)
-                        .taggedMetricRegistry(params.taggedMetrics())
-                        .build();
-
-                return new AbstractStickyChannels() {
-                    @Override
-                    public Channel getSingleHostChannel() {
-                        return stickyEndpointChannels.getStickyChannel();
-                    }
-                };
+            if (singleHostChannels.size() == 1) {
+                return () -> singleHostChannels.get(0);
```

Comment:
:+1: this refactor to centralize zero and one uri cases makes the code a lot easier to read, thanks!